src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -95147,5 +95147,59 @@ public override void M<U>(B x, U y)
             // Per https://github.com/dotnet/roslyn/issues/35075 errors should be expected
             c.VerifyDiagnostics();
         }
+
+        [Fact, WorkItem(34827, "https://github.com/dotnet/roslyn/issues/34827")]
+        public void Verify34827()
+        {
+            string source = @"
+using System.Collections.Generic;
+class C
+{
+    void M()
+    {
+        IEnumerable<object> x1 = new[] { ""string"", null }; // 1
+        IEnumerable<object?> x2 = new[] { ""string"", null };
```

Comment:
>, null [](start = 52, length = 6)

Consider dropping `null` from the cases where the target has a nullable element type.